### PR TITLE
Search Sept 2024 Preview regen

### PR DESCRIPTION
**Pull request by jairmyree**
This PR regenerates the Search SDK in preparation for the Sept 2024 preview release.

Summary of the changes: 

- VectorSearchCompression.truncationDimension has been added to enable truncation of the vector field value
- SplitSkill adds support for tokenization, this is a new TextSplitMode value and a new property for configuring Azure OpenAI tokinization configurations
- VectorQuery.filterOverride has been added to allow for vector queries to override the broader SearchRequest.filter to allow for vector queries to have more specific configurations.
- DocumentDebugInfo is extended with vector scores for the result.

### CODE REVIEW

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SplitSkillEncoderModelName.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+/**
+ * A value indicating which tokenizer to use.
+ */
+public final class SplitSkillEncoderModelName extends ExpandableStringEnum<SplitSkillEncoderModelName> {
+    /**
+     * Refers to a base model trained with a 50,000 token vocabulary, often used in general natural language processing
+     * tasks.
+     */
+    public static final SplitSkillEncoderModelName R50KBASE = fromString("r50k_base");
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SplitSkillEncoderModelName.java:20]: We're going to get feedback on the naming here (I know this is what Autorest generates). We'll likely need customizations to turn this into `R_50K_BASE`/.

File: sdk/search/azure-search-documents/swagger/README.md [hunk 2]
```diff
@@ -449,3 +449,6 @@ directive:
     $.OcrSkillLineEnding["x-ms-client-name"] = "OcrLineEnding";
     $.OcrSkillLineEnding["x-ms-enum"].name = "OcrLineEnding";
 ```
+### Remove `ClientRequestIdParameter` parameter
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/swagger/README.md:452]: Maybe let's do this in another PR?

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/Speller.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+package com.azure.search.documents.models;
+
+import com.azure.core.util.ExpandableStringEnum;
+import java.util.Collection;
+
+/**
+ * Defines values for Speller.
+ */
+public final class Speller extends ExpandableStringEnum<Speller> {
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/Speller.java:15]: Is this class the same as `QuerySpellerType`? Is this a Swagger transform issue happening where both are being generated?

Comment (jairmyree) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/Speller.java:15]: @alzimmermsft it does seem to be the same as `QuerySpellerType` but I couldn't find any swagger transform related to either. I'll add a transform to rename `Speller` to `QuerySpellerType`.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/QueryDebugMode.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+public final class QueryDebugMode extends ExpandableStringEnum<QueryDebugMode> {
+    /**
+     * No query debugging information will be returned.
+     */
+    public static final QueryDebugMode DISABLED = fromString("disabled");
+
+    /**
+     * Allows the user to further explore their reranked results.
+     */
+    public static final QueryDebugMode SEMANTIC = fromString("semantic");
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/QueryDebugMode.java:24]: This is missing `vector`

File: sdk/search/azure-search-documents/swagger/README.md [hunk 5]
```diff
@@ -87,10 +87,10 @@ These settings apply only when `--tag=searchindex` is specified on the command l
 ``` yaml $(tag) == 'searchindex'
 namespace: com.azure.search.documents
 input-file:
-- https://raw.githubusercontent.com/Azure/azure-rest-api-specs/dc27f9b32787533cd4d07fe0de5245f2f8354dbe/specification/search/data-plane/Azure.Search/stable/2024-07-01/searchindex.json
+- https://raw.githubusercontent.com/Azure/azure-rest-api-specs/4b7fbd8b842b509a0330f20260821dd844328dff/specification/search/data-plane/Azure.Search/preview/2024-09-01-preview/searchindex.json
 models-subpackage: models
 custom-types-subpackage: implementation.models
-custom-types: AutocompleteRequest,IndexAction,IndexBatch,RequestOptions,SearchDocumentsResult,SearchErrorException,SearchOptions,SearchRequest,SearchResult,SuggestDocumentsResult,SuggestRequest,SuggestResult,ErrorAdditionalInfo,ErrorDetail,ErrorResponse,ErrorResponseException
+custom-types: AutocompleteRequest,IndexAction,IndexBatch,RequestOptions,SearchDocumentsResult,SearchErrorException,SearchOptions,SearchRequest,SearchResult,SuggestDocumentsResult,SuggestRequest,SuggestResult,ErrorAdditionalInfo,ErrorDetail,ErrorResponse,ErrorResponseException,Speller
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/swagger/README.md:93]: For the `Speller` and `QuerySpeller` naming there was a Swagger transform for this before that was

````
### Rename Speller to QuerySpellerType
``` yaml $(java)
directive:
  - from: swagger-document
    where: $.paths["/docs"].get.parameters
    transform: >
      $.find(p => p.name === "speller")["x-ms-enum"].name = "QuerySpellerType";
```

### Rename Speller to QuerySpellerType
``` yaml $(tag) == 'searchindex'
directive:
  - from: swagger-document
    where: $.definitions
    transform: >
      $.Speller["x-ms-enum"].name = "QuerySpellerType";
```
````
